src/toxcore/tcp/server/server.rs

```diff
@@ -323,37 +368,56 @@ impl Server {
 
 #[cfg(test)]
 mod tests {
+    extern crate rand;
```

Comment:
Oh cmon. These kind of tests do not require randomness.

Reply:
Yes, but I added it for method 'add_random_client'. In other case I had to pass IpAddr there every time I need to add client because this method is used to add multiple clients. I can remove randomness here but it will increase boilerplate.

Reply:
And perhaps in this case we should get rid of gen_keypair() as well :)

Reply:
There is a non 0 probability that 2 calls for a rand addr will return **the same addr**. And for now the code is unspecified what will happen if the client connects with the same addr.